deeplinkdispatch-base/src/main/java/com/airbnb/deeplinkdispatch/Parser.java

```diff
@@ -1,8 +1,35 @@
 package com.airbnb.deeplinkdispatch;
 
+import java.util.List;
+
 /**
  * Parser classes are used for keeping a registry of deep links and dispatching them from Intents.
  */
-public interface Parser {
-  DeepLinkEntry parseUri(String uri);
+public abstract class Parser {
```

Comment:
this could be a good time to introduce kotlin to this project and make this a kotlin class. that would also help make clear that this is a read only list

Reply:
That's true, but I don't want to do that right now.